Kconfig: Expose gnrc/sixlowpan configurations

PeterKietzmann opened the pull request:
### Contribution description
This PR moves the configuration macros of `gnrx/sixlowpan` found [here](https://github.com/RIOT-OS/RIOT/blob/master/sys/include/net/gnrc/sixlowpan/config.h) to the `CONFIG_` namespace, and exposes them to Kconfig for individual modules.

As agreed offline with @leandrolanzieri I did not include the default thread priority parameter because this should be defined by the platform, if not changed manually. We might need a better solution to this in future.

Furthermore, I did not include parameters that relate to selective fragment recovery. As agreed with @miri64, these should be changed in #12648.

### Testing procedure
- `examples/gnrc_networking` should still work as usual.
- `tests/gnrc_sixlowpan_frag` should still work as usual.
- `tests/gnrc_sixlowpan_iphc_w_vrb` should still work as usual.
- `tests/unittests/tests-gnrc_sixlowpan_frag_vrb` should still work as usual.
- You should be able to configure the module's parameters via `make menuconfig` in all applications.

### Issues/PRs references
Part of #12888

leandrolanzieri commented on sys/net/gnrc/network_layer/sixlowpan/nd/Kconfig:
```diff
@@ -0,0 +1,26 @@
+
+if KCONFIG_MODULE_GNRC_SIXLOWPAN_ND
+
+config GNRC_SIXLOWPAN_ND_AR_LTIME
```
Should we add a dependency on `MODULE_GNRC_IPV6_NIB`?

miri64 replied:
More on `MODULE_GNRC_SIXLOWPAN_ND` I think.

leandrolanzieri replied:
This already depends on `MODULE_GNRC_SIXLOWPAN_ND` because they are inside `if KCONFIG_MODULE_GNRC_SIXLOWPAN_ND`. My question was because the help string says:

>Only applicable with @ ref net_gnrc_ipv6_nib and @ ref
        net_gnrc_sixlowpan_nd modules.

miri64 replied:
Ah, well those kind of comments in my doc usually mean "it has no effect if the module referred to is not compiled in", so I wouldn't really call it a dependency, but more of a condition.

leandrolanzieri replied:
> it has no effect if the module referred to is not compiled in

Ok, so this means that without `gnrc_ipv6_nib` module this options is not applicable, so I think it should not show up.

leandrolanzieri commented on sys/net/gnrc/Kconfig:
```diff
@@ -10,5 +10,10 @@ menu "GNRC Network stack"
+rsource "network_layer/sixlowpan/frag/fb/Kconfig"
+rsource "network_layer/sixlowpan/frag/rb/Kconfig"
+rsource "network_layer/sixlowpan/frag/vrb/Kconfig"
+rsource "network_layer/sixlowpan/nd/Kconfig"
```
What about including all these from `sixlowpan/Kconfig`? Also `sixlowpan/frag/*/Kconfig` can be included from a `sixlowpan/frag/Kconfig`.

PeterKietzmann commented on sys/net/gnrc/network_layer/sixlowpan/frag/rb/Kconfig:
```diff
@@ -0,0 +1,44 @@
+    int "Timeout for reassembly buffer entries in microseconds"
+    default 3000000
+
+config GNRC_SIXLOWPAN_FRAG_RBUF_AGGRESSIVE_OVERRIDE
```
Please have a look at #13129

PeterKietzmann commented:
I squashed the commit that has fixed one symbol discription. Furthermore, I've added one commit that removes the duplication of `CONFIG_GNRC_SIXLOWPAN_ND_AR_LTIME` and an other commit that moves the Kconfig includes to a recursive approach. Please let me know if these can be squashed. Finally, we should agree on a solution in #13129 that we apply here afterwards before we can merge.

PeterKietzmann commented:
I've added #13129 here and adapted it to this PRs needs. After #13129 got merged, this one needs a rebase. Furthermore, there a some commits that need to be squashed. @leandrolanzieri , @fjmolinas I have to leave soon. If you want take over to merge this PR before feature freeze, feel free!

PeterKietzmann commented:
THis PR should be good now after #13129 got merge and a rebase. Can I squash?

leandrolanzieri commented on sys/net/gnrc/network_layer/sixlowpan/frag/Kconfig:
```diff
@@ -0,0 +1,9 @@
+#
+rsource "fb/Kconfig"
+rsource "rb/Kconfig"
+rsource "vrb/Kconfig"
```
These should depend on `MODULE_GNRC_IPV6_SIXLOWPAN_FRAG`:
```Kconfig
if MODULE_GNRC_IPV6_SIXLOWPAN_FRAG

rsource "fb/Kconfig"
rsource "rb/Kconfig"
rsource "vrb/Kconfig"

endif # MODULE_GNRC_IPV6_SIXLOWPAN_FRAG
```
(Note that this is not a 'conditional include though, it adds this dependency to all the symbols inside).

PeterKietzmann commented:
@leandrolanzieri, addressed your comments. Squash?

miri64 commented on sys/net/gnrc/network_layer/sixlowpan/frag/rb/Kconfig:
```diff
@@ -0,0 +1,42 @@
+    help
+        When not set, it will cause the reassembly buffer to
+        override the oldest entry no matter what. When set, only the oldest
+        entry that is older than @ref CONFIG_GNRC_SIXLOWPAN_FRAG_RBUF_TIMEOUT_US
```
Does Kconfig also parse this `@ref`?

leandrolanzieri replied:
No, right now Kconfig files are not parsed for documentation. This is only kept so we don't lose track of references when documentation is generated from them in the future.

PeterKietzmann replied:
AFAIK it does not. But as we plan to parse the doxygen documentation from Kconfig files in the long run, we decided to keep these references in.

miri64 replied:
Ok

leandrolanzieri commented on sys/include/net/gnrc/sixlowpan/config.h:
```diff
@@ -95,7 +95,7 @@ extern "C" {
  */
 #ifdef DOXYGEN
-#define GNRC_SIXLOWPAN_FRAG_RBUF_DO_NOT_OVERRIDE
+#define CONFIG_GNRC_SIXLOWPAN_FRAG_RBUF_DO_NOT_OVERRIDE
```
In the documentation of this macro `GNRC_SIXLOWPAN_FRAG_RBUF_TIMEOUT_US` -> `CONFIG_GNRC_SIXLOWPAN_FRAG_RBUF_TIMEOUT_US`.

leandrolanzieri commented on sys/net/gnrc/network_layer/sixlowpan/frag/vrb/Kconfig:
```diff
@@ -0,0 +1,27 @@
+    int "Size of the virtual reassembly buffer"
+    default 16
+    help
+        Only applicable with(@ref net_gnrc_sixlowpan_frag_vrb) module,
```
One can already tell the dependency from this file, I don't think we need this line.

leandrolanzieri commented on sys/net/gnrc/network_layer/sixlowpan/frag/vrb/Kconfig:
```diff
@@ -0,0 +1,27 @@
+    help
+        Only applicable with(@ref net_gnrc_sixlowpan_frag_vrb) module,
+        but has also a direct influence on the number of available
+        gnrc_sixlowpan_frag_rb_int_t entries.
```
@miri64 is this still valid?

miri64 replied:
Jain. The dependency is introduced in #11068. So rather it is not *yet* valid.

leandrolanzieri replied:
Ok!

leandrolanzieri commented:
@PeterKietzmann Code looks good, from my side you can squash. I will be running the tests now.

leandrolanzieri commented:
### Test results

<details><summary><b>tests/gnrc_sixlowpan_frag (without Kconfig)</b></summary>

```
Building application "tests_gnrc_sixlowpan_frag" for "samr21-xpro" with MCU "samd21".

"make" -C /home/leandro/Work/RIOT/boards/samr21-xpro
"make" -C /home/leandro/Work/RIOT/core
"make" -C /home/leandro/Work/RIOT/cpu/samd21
"make" -C /home/leandro/Work/RIOT/cpu/cortexm_common
"make" -C /home/leandro/Work/RIOT/cpu/cortexm_common/periph
"make" -C /home/leandro/Work/RIOT/cpu/sam0_common
"make" -C /home/leandro/Work/RIOT/cpu/sam0_common/periph
"make" -C /home/leandro/Work/RIOT/cpu/samd21/periph
"make" -C /home/leandro/Work/RIOT/drivers
"make" -C /home/leandro/Work/RIOT/drivers/periph_common
"make" -C /home/leandro/Work/RIOT/sys
"make" -C /home/leandro/Work/RIOT/sys/div
"make" -C /home/leandro/Work/RIOT/sys/embunit
"make" -C /home/leandro/Work/RIOT/sys/fmt
"make" -C /home/leandro/Work/RIOT/sys/isrpipe
"make" -C /home/leandro/Work/RIOT/sys/net/crosslayer/inet_csum
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netapi
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif/hdr
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netreg
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/fb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/rb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pkt
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pktbuf
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pktbuf_static
"make" -C /home/leandro/Work/RIOT/sys/net/link_layer/l2util
"make" -C /home/leandro/Work/RIOT/sys/net/netif
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/ipv6/addr
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/ipv6/hdr
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/sixlowpan
"make" -C /home/leandro/Work/RIOT/sys/newlib_syscalls_default
"make" -C /home/leandro/Work/RIOT/sys/pm_layered
"make" -C /home/leandro/Work/RIOT/sys/stdio_uart
"make" -C /home/leandro/Work/RIOT/sys/test_utils/interactive_sync
"make" -C /home/leandro/Work/RIOT/sys/tsrb
"make" -C /home/leandro/Work/RIOT/sys/xtimer
   text	  data	   bss	   dec	   hex	filename
  21944	   516	  5368	 27828	  6cb4	/home/leandro/Work/RIOT/tests/gnrc_sixlowpan_frag/bin/samr21-xpro/tests_gnrc_sixlowpan_frag.elf
/home/leandro/Work/RIOT/dist/tools/edbg/edbg --serial ATML2127031800002743 --serial ATML2127031800002743  --target atmel_cm0p --verbose --file /home/leandro/Work/RIOT/tests/gnrc_sixlowpan_frag/bin/samr21-xpro/tests_gnrc_sixlowpan_frag.bin --verify || /home/leandro/Work/RIOT/dist/tools/edbg/edbg --serial ATML2127031800002743 --serial ATML2127031800002743  --target atmel_cm0p --verbose --file /home/leandro/Work/RIOT/tests/gnrc_sixlowpan_frag/bin/samr21-xpro/tests_gnrc_sixlowpan_frag.bin --verify --program
Debugger: ATMEL EDBG CMSIS-DAP ATML2127031800002743 03.25.01B6 (S)
Clock frequency: 16.0 MHz
Target: SAM R21G18 (Rev B)
Verification...
at address 0x4 expected 0x01, read 0x45
Error: verification failed
Debugger: ATMEL EDBG CMSIS-DAP ATML2127031800002743 03.25.01B6 (S)
Clock frequency: 16.0 MHz
Target: SAM R21G18 (Rev B)
Programming........................................................................................... done.
Verification........................................................................................... done.
r
/home/leandro/Work/RIOT/dist/tools/pyterm/pyterm -p "/dev/ttyACM0" -b "115200" --noprefix --no-repeat-command-on-empty-line
Twisted not available, please install it if you want to use pyterm's JSON capabilities
Connect to serial port /dev/ttyACM0
Welcome to pyterm!
Type '/exit' to exit.
READY
s
START
.............
OK (13 tests)
```
</details>

<details><summary><b>tests/gnrc_sixlowpan_frag (with Kconfig defaults)</b></summary>

```
 > env BOARD=samr21-xpro SERIAL=ATML2127031800002743 make menuconfig
Using default symbol values (no '/home/leandro/Work/RIOT/tests/gnrc_sixlowpan_frag/bin/samr21-xpro/generated/merged.config')
No changes to save (for '/home/leandro/Work/RIOT/tests/gnrc_sixlowpan_frag/bin/samr21-xpro/generated/merged.config')

 > env BOARD=samr21-xpro SERIAL=ATML2127031800002743 make all flash test
Building application "tests_gnrc_sixlowpan_frag" for "samr21-xpro" with MCU "samd21".

"make" -C /home/leandro/Work/RIOT/boards/samr21-xpro
"make" -C /home/leandro/Work/RIOT/core
"make" -C /home/leandro/Work/RIOT/cpu/samd21
"make" -C /home/leandro/Work/RIOT/cpu/cortexm_common
"make" -C /home/leandro/Work/RIOT/cpu/cortexm_common/periph
"make" -C /home/leandro/Work/RIOT/cpu/sam0_common
"make" -C /home/leandro/Work/RIOT/cpu/sam0_common/periph
"make" -C /home/leandro/Work/RIOT/cpu/samd21/periph
"make" -C /home/leandro/Work/RIOT/drivers
"make" -C /home/leandro/Work/RIOT/drivers/periph_common
"make" -C /home/leandro/Work/RIOT/sys
"make" -C /home/leandro/Work/RIOT/sys/div
"make" -C /home/leandro/Work/RIOT/sys/embunit
"make" -C /home/leandro/Work/RIOT/sys/fmt
"make" -C /home/leandro/Work/RIOT/sys/isrpipe
"make" -C /home/leandro/Work/RIOT/sys/net/crosslayer/inet_csum
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netapi
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif/hdr
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netreg
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/fb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/rb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pkt
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pktbuf
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pktbuf_static
"make" -C /home/leandro/Work/RIOT/sys/net/link_layer/l2util
"make" -C /home/leandro/Work/RIOT/sys/net/netif
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/ipv6/addr
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/ipv6/hdr
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/sixlowpan
"make" -C /home/leandro/Work/RIOT/sys/newlib_syscalls_default
"make" -C /home/leandro/Work/RIOT/sys/pm_layered
"make" -C /home/leandro/Work/RIOT/sys/stdio_uart
"make" -C /home/leandro/Work/RIOT/sys/test_utils/interactive_sync
"make" -C /home/leandro/Work/RIOT/sys/tsrb
"make" -C /home/leandro/Work/RIOT/sys/xtimer
   text	  data	   bss	   dec	   hex	filename
  21944	   516	  5368	 27828	  6cb4	/home/leandro/Work/RIOT/tests/gnrc_sixlowpan_frag/bin/samr21-xpro/tests_gnrc_sixlowpan_frag.elf
/home/leandro/Work/RIOT/dist/tools/edbg/edbg --serial ATML2127031800002743 --serial ATML2127031800002743  --target atmel_cm0p --verbose --file /home/leandro/Work/RIOT/tests/gnrc_sixlowpan_frag/bin/samr21-xpro/tests_gnrc_sixlowpan_frag.bin --verify || /home/leandro/Work/RIOT/dist/tools/edbg/edbg --serial ATML2127031800002743 --serial ATML2127031800002743  --target atmel_cm0p --verbose --file /home/leandro/Work/RIOT/tests/gnrc_sixlowpan_frag/bin/samr21-xpro/tests_gnrc_sixlowpan_frag.bin --verify --program
Debugger: ATMEL EDBG CMSIS-DAP ATML2127031800002743 03.25.01B6 (S)
Clock frequency: 16.0 MHz
Target: SAM R21G18 (Rev B)
Verification........................................................................................... done.
r
/home/leandro/Work/RIOT/dist/tools/pyterm/pyterm -p "/dev/ttyACM0" -b "115200" --noprefix --no-repeat-command-on-empty-line
Twisted not available, please install it if you want to use pyterm's JSON capabilities
Connect to serial port /dev/ttyACM0
Welcome to pyterm!
Type '/exit' to exit.
READY
s
START
.............
OK (13 tests)

 ~/W/R/t/gnrc_sixlowpan_frag   pr_kconf_6lo   
```
</details>

---

<details><summary><b>tests/gnrc_sixlowpan_iphc_w_vrb (without Kconfig)</b></summary>

```
> env BOARD=samr21-xpro SERIAL=ATML2127031800002743 make clean

> env BOARD=samr21-xpro SERIAL=ATML2127031800002743 make all flash test
Building application "tests_gnrc_sixlowpan_iphc_w_vrb" for "samr21-xpro" with MCU "samd21".

"make" -C /home/leandro/Work/RIOT/boards/samr21-xpro
"make" -C /home/leandro/Work/RIOT/core
"make" -C /home/leandro/Work/RIOT/cpu/samd21
"make" -C /home/leandro/Work/RIOT/cpu/cortexm_common
"make" -C /home/leandro/Work/RIOT/cpu/cortexm_common/periph
"make" -C /home/leandro/Work/RIOT/cpu/sam0_common
"make" -C /home/leandro/Work/RIOT/cpu/sam0_common/periph
"make" -C /home/leandro/Work/RIOT/cpu/samd21/periph
"make" -C /home/leandro/Work/RIOT/drivers
"make" -C /home/leandro/Work/RIOT/drivers/netdev_ieee802154
"make" -C /home/leandro/Work/RIOT/drivers/periph_common
"make" -C /home/leandro/Work/RIOT/sys
"make" -C /home/leandro/Work/RIOT/sys/auto_init
"make" -C /home/leandro/Work/RIOT/sys/div
"make" -C /home/leandro/Work/RIOT/sys/embunit
"make" -C /home/leandro/Work/RIOT/sys/evtimer
"make" -C /home/leandro/Work/RIOT/sys/fmt
"make" -C /home/leandro/Work/RIOT/sys/isrpipe
"make" -C /home/leandro/Work/RIOT/sys/luid
"make" -C /home/leandro/Work/RIOT/sys/net/crosslayer/inet_csum
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netapi
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif/hdr
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif/ieee802154
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netreg
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/icmpv6
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/ipv6
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/ipv6/hdr
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/ipv6/nib
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/ndp
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/ctx
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/fb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/rb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/vrb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/iphc
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/nd
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pkt
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pktbuf
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pktbuf_static
"make" -C /home/leandro/Work/RIOT/sys/net/link_layer/ieee802154
"make" -C /home/leandro/Work/RIOT/sys/net/link_layer/l2util
"make" -C /home/leandro/Work/RIOT/sys/net/netdev_test
"make" -C /home/leandro/Work/RIOT/sys/net/netif
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/icmpv6
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/ipv6/addr
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/ipv6/hdr
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/sixlowpan
"make" -C /home/leandro/Work/RIOT/sys/newlib_syscalls_default
"make" -C /home/leandro/Work/RIOT/sys/od
"make" -C /home/leandro/Work/RIOT/sys/pm_layered
"make" -C /home/leandro/Work/RIOT/sys/random
"make" -C /home/leandro/Work/RIOT/sys/random/tinymt32
"make" -C /home/leandro/Work/RIOT/sys/stdio_uart
"make" -C /home/leandro/Work/RIOT/sys/test_utils/interactive_sync
"make" -C /home/leandro/Work/RIOT/sys/tsrb
"make" -C /home/leandro/Work/RIOT/sys/xtimer
   text	  data	   bss	   dec	   hex	filename
  43680	   148	 11144	 54972	  d6bc	/home/leandro/Work/RIOT/tests/gnrc_sixlowpan_iphc_w_vrb/bin/samr21-xpro/tests_gnrc_sixlowpan_iphc_w_vrb.elf
/home/leandro/Work/RIOT/dist/tools/edbg/edbg --serial ATML2127031800002743 --serial ATML2127031800002743  --target atmel_cm0p --verbose --file /home/leandro/Work/RIOT/tests/gnrc_sixlowpan_iphc_w_vrb/bin/samr21-xpro/tests_gnrc_sixlowpan_iphc_w_vrb.bin --verify || /home/leandro/Work/RIOT/dist/tools/edbg/edbg --serial ATML2127031800002743 --serial ATML2127031800002743  --target atmel_cm0p --verbose --file /home/leandro/Work/RIOT/tests/gnrc_sixlowpan_iphc_w_vrb/bin/samr21-xpro/tests_gnrc_sixlowpan_iphc_w_vrb.bin --verify --program
Debugger: ATMEL EDBG CMSIS-DAP ATML2127031800002743 03.25.01B6 (S)
Clock frequency: 16.0 MHz
Target: SAM R21G18 (Rev B)
Verification...
at address 0x4 expected 0x45, read 0x01
Error: verification failed
Debugger: ATMEL EDBG CMSIS-DAP ATML2127031800002743 03.25.01B6 (S)
Clock frequency: 16.0 MHz
Target: SAM R21G18 (Rev B)
Programming............................................................................................................................................................................... done.
Verification............................................................................................................................................................................... done.
r
/home/leandro/Work/RIOT/dist/tools/pyterm/pyterm -p "/dev/ttyACM0" -b "115200" --noprefix --no-repeat-command-on-empty-line
Twisted not available, please install it if you want to use pyterm's JSON capabilities
Connect to serial port /dev/ttyACM0
Welcome to pyterm!
Type '/exit' to exit.
READY
s
START
main(): This is RIOT! (Version: 2020.04-devel-38-g96ff9-pr_kconf_6lo)
....
OK (4 tests)
```
</details>

<details><summary><b>tests/gnrc_sixlowpan_iphc_w_vrb (with Kconfig defaults)</b></summary>

```
> env BOARD=samr21-xpro SERIAL=ATML2127031800002743 make menuconfig
Using default symbol values (no '/home/leandro/Work/RIOT/tests/gnrc_sixlowpan_iphc_w_vrb/bin/samr21-xpro/generated/merged.config')
No changes to save (for '/home/leandro/Work/RIOT/tests/gnrc_sixlowpan_iphc_w_vrb/bin/samr21-xpro/generated/merged.config')

> env BOARD=samr21-xpro SERIAL=ATML2127031800002743 make all flash test
Building application "tests_gnrc_sixlowpan_iphc_w_vrb" for "samr21-xpro" with MCU "samd21".

"make" -C /home/leandro/Work/RIOT/boards/samr21-xpro
"make" -C /home/leandro/Work/RIOT/core
"make" -C /home/leandro/Work/RIOT/cpu/samd21
"make" -C /home/leandro/Work/RIOT/cpu/cortexm_common
"make" -C /home/leandro/Work/RIOT/cpu/cortexm_common/periph
"make" -C /home/leandro/Work/RIOT/cpu/sam0_common
"make" -C /home/leandro/Work/RIOT/cpu/sam0_common/periph
"make" -C /home/leandro/Work/RIOT/cpu/samd21/periph
"make" -C /home/leandro/Work/RIOT/drivers
"make" -C /home/leandro/Work/RIOT/drivers/netdev_ieee802154
"make" -C /home/leandro/Work/RIOT/drivers/periph_common
"make" -C /home/leandro/Work/RIOT/sys
"make" -C /home/leandro/Work/RIOT/sys/auto_init
"make" -C /home/leandro/Work/RIOT/sys/div
"make" -C /home/leandro/Work/RIOT/sys/embunit
"make" -C /home/leandro/Work/RIOT/sys/evtimer
"make" -C /home/leandro/Work/RIOT/sys/fmt
"make" -C /home/leandro/Work/RIOT/sys/isrpipe
"make" -C /home/leandro/Work/RIOT/sys/luid
"make" -C /home/leandro/Work/RIOT/sys/net/crosslayer/inet_csum
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netapi
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif/hdr
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif/ieee802154
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netreg
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/icmpv6
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/ipv6
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/ipv6/hdr
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/ipv6/nib
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/ndp
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/ctx
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/fb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/rb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/vrb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/iphc
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/nd
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pkt
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pktbuf
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pktbuf_static
"make" -C /home/leandro/Work/RIOT/sys/net/link_layer/ieee802154
"make" -C /home/leandro/Work/RIOT/sys/net/link_layer/l2util
"make" -C /home/leandro/Work/RIOT/sys/net/netdev_test
"make" -C /home/leandro/Work/RIOT/sys/net/netif
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/icmpv6
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/ipv6/addr
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/ipv6/hdr
"make" -C /home/leandro/Work/RIOT/sys/net/network_layer/sixlowpan
"make" -C /home/leandro/Work/RIOT/sys/newlib_syscalls_default
"make" -C /home/leandro/Work/RIOT/sys/od
"make" -C /home/leandro/Work/RIOT/sys/pm_layered
"make" -C /home/leandro/Work/RIOT/sys/random
"make" -C /home/leandro/Work/RIOT/sys/random/tinymt32
"make" -C /home/leandro/Work/RIOT/sys/stdio_uart
"make" -C /home/leandro/Work/RIOT/sys/test_utils/interactive_sync
"make" -C /home/leandro/Work/RIOT/sys/tsrb
"make" -C /home/leandro/Work/RIOT/sys/xtimer
   text	  data	   bss	   dec	   hex	filename
  43680	   148	 11144	 54972	  d6bc	/home/leandro/Work/RIOT/tests/gnrc_sixlowpan_iphc_w_vrb/bin/samr21-xpro/tests_gnrc_sixlowpan_iphc_w_vrb.elf
/home/leandro/Work/RIOT/dist/tools/edbg/edbg --serial ATML2127031800002743 --serial ATML2127031800002743  --target atmel_cm0p --verbose --file /home/leandro/Work/RIOT/tests/gnrc_sixlowpan_iphc_w_vrb/bin/samr21-xpro/tests_gnrc_sixlowpan_iphc_w_vrb.bin --verify || /home/leandro/Work/RIOT/dist/tools/edbg/edbg --serial ATML2127031800002743 --serial ATML2127031800002743  --target atmel_cm0p --verbose --file /home/leandro/Work/RIOT/tests/gnrc_sixlowpan_iphc_w_vrb/bin/samr21-xpro/tests_gnrc_sixlowpan_iphc_w_vrb.bin --verify --program
Debugger: ATMEL EDBG CMSIS-DAP ATML2127031800002743 03.25.01B6 (S)
Clock frequency: 16.0 MHz
Target: SAM R21G18 (Rev B)
Verification............................................................................................................................................................................... done.
r
/home/leandro/Work/RIOT/dist/tools/pyterm/pyterm -p "/dev/ttyACM0" -b "115200" --noprefix --no-repeat-command-on-empty-line
Twisted not available, please install it if you want to use pyterm's JSON capabilities
Connect to serial port /dev/ttyACM0
Welcome to pyterm!
Type '/exit' to exit.
READY
s
START
main(): This is RIOT! (Version: 2020.04-devel-38-g96ff9-pr_kconf_6lo)
....
OK (4 tests)
```
</details>

---

<details><summary><b>tests/unittests/gnrc_sixlowpan_frag_vrb (without Kconfig)</b></summary>

```
 > env BOARD=samr21-xpro SERIAL=ATML2127031800002743 make clean
 
> env BOARD=samr21-xpro SERIAL=ATML2127031800002743 make tests-gnrc_sixlowpan_frag_vrb flash test
Building application "tests_unittests" for "samr21-xpro" with MCU "samd21".

"make" -C /home/leandro/Work/RIOT/boards/samr21-xpro
"make" -C /home/leandro/Work/RIOT/core
"make" -C /home/leandro/Work/RIOT/cpu/samd21
"make" -C /home/leandro/Work/RIOT/cpu/cortexm_common
"make" -C /home/leandro/Work/RIOT/cpu/cortexm_common/periph
"make" -C /home/leandro/Work/RIOT/cpu/sam0_common
"make" -C /home/leandro/Work/RIOT/cpu/sam0_common/periph
"make" -C /home/leandro/Work/RIOT/cpu/samd21/periph
"make" -C /home/leandro/Work/RIOT/drivers
"make" -C /home/leandro/Work/RIOT/drivers/periph_common
"make" -C /home/leandro/Work/RIOT/sys
"make" -C /home/leandro/Work/RIOT/sys/div
"make" -C /home/leandro/Work/RIOT/sys/embunit
"make" -C /home/leandro/Work/RIOT/sys/fmt
"make" -C /home/leandro/Work/RIOT/sys/isrpipe
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netapi
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif/hdr
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netreg
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/fb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/vrb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pkt
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pktbuf
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pktbuf_static
"make" -C /home/leandro/Work/RIOT/sys/net/link_layer/l2util
"make" -C /home/leandro/Work/RIOT/sys/net/netif
"make" -C /home/leandro/Work/RIOT/sys/newlib_syscalls_default
"make" -C /home/leandro/Work/RIOT/sys/pm_layered
"make" -C /home/leandro/Work/RIOT/sys/stdio_uart
"make" -C /home/leandro/Work/RIOT/sys/test_utils/interactive_sync
"make" -C /home/leandro/Work/RIOT/sys/tsrb
"make" -C /home/leandro/Work/RIOT/sys/xtimer
"make" -C /home/leandro/Work/RIOT/tests/unittests/tests-gnrc_sixlowpan_frag_vrb
   text	  data	   bss	   dec	   hex	filename
  13896	   148	  3372	 17416	  4408	/home/leandro/Work/RIOT/tests/unittests/bin/samr21-xpro/tests_unittests.elf
/home/leandro/Work/RIOT/dist/tools/edbg/edbg --serial ATML2127031800002743 --serial ATML2127031800002743  --target atmel_cm0p --verbose --file /home/leandro/Work/RIOT/tests/unittests/bin/samr21-xpro/tests_unittests.bin --verify || /home/leandro/Work/RIOT/dist/tools/edbg/edbg --serial ATML2127031800002743 --serial ATML2127031800002743  --target atmel_cm0p --verbose --file /home/leandro/Work/RIOT/tests/unittests/bin/samr21-xpro/tests_unittests.bin --verify --program
Debugger: ATMEL EDBG CMSIS-DAP ATML2127031800002743 03.25.01B6 (S)
Clock frequency: 16.0 MHz
Target: SAM R21G18 (Rev B)
Verification...
at address 0x4 expected 0xc1, read 0x45
Error: verification failed
Debugger: ATMEL EDBG CMSIS-DAP ATML2127031800002743 03.25.01B6 (S)
Clock frequency: 16.0 MHz
Target: SAM R21G18 (Rev B)
Programming.......................................................... done.
Verification.......................................................... done.
r
/home/leandro/Work/RIOT/dist/tools/pyterm/pyterm -p "/dev/ttyACM0" -b "115200" --noprefix --no-repeat-command-on-empty-line
Twisted not available, please install it if you want to use pyterm's JSON capabilities
Connect to serial port /dev/ttyACM0
Welcome to pyterm!
Type '/exit' to exit.
READY
s
START
.......
OK (7 tests)
```
</details>

<details><summary><b>tests/unittests/gnrc_sixlowpan_frag_vrb (with Kconfig defaults)</b></summary>

```
> env BOARD=samr21-xpro SERIAL=ATML2127031800002743 make menuconfig
Using default symbol values (no '/home/leandro/Work/RIOT/tests/unittests/bin/samr21-xpro/generated/merged.config')
No changes to save (for '/home/leandro/Work/RIOT/tests/unittests/bin/samr21-xpro/generated/merged.config')

> env BOARD=samr21-xpro SERIAL=ATML2127031800002743 make tests-gnrc_sixlowpan_frag_vrb flash test
Building application "tests_unittests" for "samr21-xpro" with MCU "samd21".

"make" -C /home/leandro/Work/RIOT/boards/samr21-xpro
"make" -C /home/leandro/Work/RIOT/core
"make" -C /home/leandro/Work/RIOT/cpu/samd21
"make" -C /home/leandro/Work/RIOT/cpu/cortexm_common
"make" -C /home/leandro/Work/RIOT/cpu/cortexm_common/periph
"make" -C /home/leandro/Work/RIOT/cpu/sam0_common
"make" -C /home/leandro/Work/RIOT/cpu/sam0_common/periph
"make" -C /home/leandro/Work/RIOT/cpu/samd21/periph
"make" -C /home/leandro/Work/RIOT/drivers
"make" -C /home/leandro/Work/RIOT/drivers/periph_common
"make" -C /home/leandro/Work/RIOT/sys
"make" -C /home/leandro/Work/RIOT/sys/div
"make" -C /home/leandro/Work/RIOT/sys/embunit
"make" -C /home/leandro/Work/RIOT/sys/fmt
"make" -C /home/leandro/Work/RIOT/sys/isrpipe
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netapi
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netif/hdr
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/netreg
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/fb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/network_layer/sixlowpan/frag/vrb
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pkt
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pktbuf
"make" -C /home/leandro/Work/RIOT/sys/net/gnrc/pktbuf_static
"make" -C /home/leandro/Work/RIOT/sys/net/link_layer/l2util
"make" -C /home/leandro/Work/RIOT/sys/net/netif
"make" -C /home/leandro/Work/RIOT/sys/newlib_syscalls_default
"make" -C /home/leandro/Work/RIOT/sys/pm_layered
"make" -C /home/leandro/Work/RIOT/sys/stdio_uart
"make" -C /home/leandro/Work/RIOT/sys/test_utils/interactive_sync
"make" -C /home/leandro/Work/RIOT/sys/tsrb
"make" -C /home/leandro/Work/RIOT/sys/xtimer
"make" -C /home/leandro/Work/RIOT/tests/unittests/tests-gnrc_sixlowpan_frag_vrb
   text	  data	   bss	   dec	   hex	filename
  13896	   148	  3372	 17416	  4408	/home/leandro/Work/RIOT/tests/unittests/bin/samr21-xpro/tests_unittests.elf
/home/leandro/Work/RIOT/dist/tools/edbg/edbg --serial ATML2127031800002743 --serial ATML2127031800002743  --target atmel_cm0p --verbose --file /home/leandro/Work/RIOT/tests/unittests/bin/samr21-xpro/tests_unittests.bin --verify || /home/leandro/Work/RIOT/dist/tools/edbg/edbg --serial ATML2127031800002743 --serial ATML2127031800002743  --target atmel_cm0p --verbose --file /home/leandro/Work/RIOT/tests/unittests/bin/samr21-xpro/tests_unittests.bin --verify --program
Debugger: ATMEL EDBG CMSIS-DAP ATML2127031800002743 03.25.01B6 (S)
Clock frequency: 16.0 MHz
Target: SAM R21G18 (Rev B)
Verification.......................................................... done.
r
/home/leandro/Work/RIOT/dist/tools/pyterm/pyterm -p "/dev/ttyACM0" -b "115200" --noprefix --no-repeat-command-on-empty-line
Twisted not available, please install it if you want to use pyterm's JSON capabilities
Connect to serial port /dev/ttyACM0
Welcome to pyterm!
Type '/exit' to exit.
READY
s
START
.......
OK (7 tests)
```
</details>

leandrolanzieri commented:
Tests were OK and code changes look good. @miri64 any comments on your side?

miri64 commented on sys/include/net/gnrc/sixlowpan/config.h:
```diff
@@ -162,16 +162,11 @@ extern "C" {
-#endif  /* GNRC_SIXLOWPAN_FRAG_VRB_SIZE */
+#ifndef CONFIG_GNRC_SIXLOWPAN_FRAG_VRB_SIZE
+#define CONFIG_GNRC_SIXLOWPAN_FRAG_VRB_SIZE        (16U)
+#endif  /* CONFIG_GNRC_SIXLOWPAN_FRAG_VRB_SIZE */
```
Note to self: adapt #11068 for this.
